src/cmd/create.rs

```diff
@@ -141,10 +150,14 @@ impl Sharded {
     }
 }
 
-fn gen_keypair(tag: KeyTag, seed_words: Option<Vec<String>>) -> Result<Keypair> {
+fn gen_keypair(
+    tag: KeyTag,
+    seed_words: Option<Vec<String>>,
+    seed_type: Option<SeedType>,
+) -> Result<Keypair> {
     match seed_words {
         Some(words) => {
-            let entropy = mnemonic_to_entropy(words)?;
+            let entropy = mnemonic_to_entropy(words, seed_type.expect("mnemonic_to_entropy"))?;
```

Comment:
Either remove the `Option` wrapper around SeedType if a SeedType is always expected, or return an error... I'm not a fan of panics

Reply:
will do.

Reply:
done. I'm not 100% happy with the way this turned out, I ended up matching on a tuple (TIL you can math on tuples in Rust), and left the default code path as a bail!, but...getting to that path wouldn't be a user error, it would be a coding mistake. Not sure if there's a better way to handle this, the only other thing I thought of was wrapping up the words and the type in its own small struct that is either all filled out or None. But, that felt even more complicated.